File: JAVA/2419. Longest Subarray With Maximum Bitwise AND.java
class Solution {
    public int longestSubarray(int[] nums) {
        int maxElement = 0;
        int maxLength = 0;
        int currentLength = 0;
        for(int num :nums) {
            if(num > maxElement) {
                maxElement = num;
                currentLength = 1;
                maxLength = 1;
            } else if(num == maxElement) {
                currentLength++;
                maxLength = Math.max(maxLength, currentLength);
            } else {
                currentLength = 0;
            }
        }
        return maxLength;
    }
}

File: JAVA/2419. Longest Subarray With Maximum Bitwise AND.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int maxAnd = *max_element(nums.begin(), nums.end());
        int len = 0;
        int res = 1;
        int n = nums.size();
        int i = 0;
        while(i < n){
            len = 0;
            while(i < n && nums[i] == maxAnd){
                res = max(res, ++len);
                i++;
            }
            i++;
        }
        return res;
    }
};
